server/pairing/handlers.go

```diff
@@ -29,145 +28,172 @@ const (
 	sessionBlocked   = "blocked"
 )
 
-func handleReceiveAccount(ps *Server) http.HandlerFunc {
+// Account handling
+
+func handleReceiveAccount(hs HandlerServer, pr PayloadReceiver) http.HandlerFunc {
 	signal.SendLocalPairingEvent(Event{Type: EventConnectionSuccess, Action: ActionPairingAccount})
-	logger := ps.GetLogger()
+	logger := hs.GetLogger()
 	return func(w http.ResponseWriter, r *http.Request) {
 		payload, err := io.ReadAll(r.Body)
 		if err != nil {
 			signal.SendLocalPairingEvent(Event{Type: EventTransferError, Error: err.Error(), Action: ActionPairingAccount})
 			logger.Error("handleReceiveAccount io.ReadAll(r.Body)", zap.Error(err))
+			http.Error(w, "error", http.StatusInternalServerError)
 			return
 		}
 		signal.SendLocalPairingEvent(Event{Type: EventTransferSuccess, Action: ActionPairingAccount})
 
-		err = ps.PayloadManager.Receive(payload)
+		err = pr.Receive(payload)
 		if err != nil {
 			signal.SendLocalPairingEvent(Event{Type: EventProcessError, Error: err.Error(), Action: ActionPairingAccount})
-			logger.Error("ps.PayloadManager.Receive(payload)", zap.Error(err))
+			logger.Error("pr.Receive(payload)", zap.Error(err))
```

Comment:
That's reasonable.

The reason the text is "pr.Receive(payload)", is because the code that generates the potential error is:

```go
err = pr.Receive(payload)
```

It just makes it easier to find if you search the codebase for the literal string. But I've had a think about what you say and I think that I will do one better. I'll change the text as you suggest, but I'll also pass in the input `payload` as an additional field to be logged.

Reply:
> The reason the text is "pr.Receive(payload)", is because the code that generates the potential error is

I thought it was the case, and grepping for the error is an important reason as you said.

> but I'll also pass in the input payload as an additional field to be logged.

:+1: